lib/posix.dart

```diff
@@ -82,8 +111,32 @@ class _PosixConditionVariable extends ConditionVariable {
   }
 
   @override
-  void wait(covariant _PosixMutex mutex) {
-    if (pthread_cond_wait(_impl, mutex._impl) != 0) {
+  void wait(covariant _PosixMutex mutex, {Duration? timeout}) {
+    if (timeout == null) {
+      if (pthread_cond_wait(_impl, mutex._impl) != 0) {
+        throw StateError('Failed to wait on a condition variable');
+      }
+    } else {
+      _timedWait(timeout, mutex);
+    }
+  }
+
+  /// Waits on a condition variable with a timeout.
+  void _timedWait(Duration timeout, _PosixMutex mutex) {
+    final abstime =
```

Comment:
The code for computing timespec for waiting should probably be shared between `_timedWait` and `_timedLock`.

Reply:
done.